File: teamup-common/src/main/kotlin/cn/net/ziqiang/teamup/backend/common/pojo/entity/Role.java
package cn.net.ziqiang.teamup.backend.common.pojo.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import javax.persistence.*;

@Entity(name = "role")
@Data
public class Role {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    Long id;

    @Column(name = "name", nullable = false, length = 30)
    @Schema(description = "角色名称")
    String name;
}

File: teamup-common/src/main/kotlin/cn/net/ziqiang/teamup/backend/common/pojo/entity/Role.kt
package cn.net.ziqiang.teamup.backend.common.pojo.entity

import javax.persistence.*

@Entity(name = "role")
class Role (
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    var id: Long? = null,

    @Column(name = "name")
    var name: String? = null,

    @Column(name = "description")
    var description: String? = null,

    @Column(name = "level")
    var level: Int? = null,

    @Column(name = "pid")
    var pid: Long? = null,
)